### PR TITLE
[WIP] Add "What is HashiCorp Vault" SEO page

### PR DESCRIPTION
**Pull request by interurban**
## Description
This PR adds content on what is hashicorp vault to the "What Is" section of the website.

## Checklist:

- [ ] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

File: themes/default/content/what-is/what-is-hashicorp-vault.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+In this article, we'll cover the key features of [HashiCorp Vault](https://www.hashicorp.com/products/vault), why secret management is important and share the use cases of how Vault integrates into various cloud architectures, enhancing the security and management of sensitive data in complex cloud environments.
+
+### Key Features of HashiCorp Vault?
+
+Docker Configs are a resource in Docker for storing non-sensitive information such as configuration files, separate from a service's image or running containers within [Docker Swarm](https://docs.docker.com/engine/swarm/) environments. This enables keeping Docker images as generic as possible without the need for bind-mounting configuration files into containers or using environment variables. Unlike [Docker Secrets](/what-is/what-are-docker-secrets), Docker Configs are not encrypted at rest and are directly mounted into the container's filesystem.
```
Comment (aaronkao) [inline themes/default/content/what-is/what-is-hashicorp-vault.md:16]: Did we take copy from the other page and not update it?

File: themes/default/content/what-is/what-is-hashicorp-vault.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+- **Edge Computing**: In edge computing scenarios, Vault can manage secrets used by edge devices and local data processing applications, ensuring security in distributed environments.
+
+### Conclusion
+
+Vault's role in each of these architectures is to provide the foundation of secure storage, strict access control, and auditing for secrets and sensitive data. When combined with Pulumi ESC, the capabilities extend even further, ensuring strong security and efficient management across your entire cloud infrastructure and target execution environments.
+
+Pulumi ESC used in combination with Hashicorp Vault enables teams to aggregate secrets and configuration from many sources, manage hierarchical collections of configuration and secrets (“environments”), and consume those configuration and secrets from a variety of different infrastructure and application services. Pulumi ESC works hand-in-hand with Pulumi IaC to simplify configuration management, but also works independently from Pulumi IaC, as a solution for managing environments, secrets and configuration for any application or infrastructure project
```
Comment (aaronkao) [inline themes/default/content/what-is/what-is-hashicorp-vault.md:83]: do we have a docs page about how to use Vault with ESC? we should link out to that.

File: themes/default/content/what-is/what-is-hashicorp-vault.md (hunk 1 above)
Comment (interurban) [reply themes/default/content/what-is/what-is-hashicorp-vault.md]: fixed

File: themes/default/content/what-is/what-is-hashicorp-vault.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+---
+title: What is Hashicorp Vault?
+meta_desc: |
+     HashiCorp Vault is a tool for securing, storing, and tightly controlling access to tokens, passwords, certificates, and encryption keys.
+
+type: what-is
+page_title: "What is Hashicorp Vault"
```
Comment (scottslowe) [inline themes/default/content/what-is/what-is-hashicorp-vault.md:7]: We (the team) landed on including the `?` here in the `page_title` field.

File: themes/default/content/what-is/what-is-hashicorp-vault.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+In this article, we'll cover the key features of [HashiCorp Vault](https://www.hashicorp.com/products/vault), why secret management is important and share the use cases of how Vault integrates into various cloud architectures, enhancing the security and management of sensitive data in complex cloud environments.
+
+### Key Features of HashiCorp Vault?
```
Comment (scottslowe) [inline themes/default/content/what-is/what-is-hashicorp-vault.md:14]: Style guide calls for sentence case for all headings (I have a PR open to fix my incorrect usage in the IaC page)

File: themes/default/content/what-is/what-is-hashicorp-vault.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+- **Secure Secret Storage**: Vault has the capability to store a wide range of key/value secrets. Before these secrets are written to persistent storage, Vault encrypts them, ensuring that direct access to the raw storage does not compromise the secrets. Vault supports various storage backends, including disk storage, Consul, among others, for versatile secret storage solutions.
```
Comment (scottslowe) [inline themes/default/content/what-is/what-is-hashicorp-vault.md:16]: I think we should also sentence case these initial in-paragraph headings, too (not specified in the style guide AFAIK)

File: themes/default/content/what-is/what-is-hashicorp-vault.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+- **Leasing and Renewal**: Secrets in Vault can have a lease associated with them. When the lease expires, the secret is automatically revoked. Clients can also renew leases to extend their validity.
+- **Revocation**: Vault can revoke not just individual secrets but entire trees of secrets. For instance, if a particular user or application is compromised, all secrets accessed by that entity can be revoked, enhancing security.
+- **Identity and Access Management**: Vault supports various authentication methods (like tokens, username/password, cloud IAM, etc.) and uses policies to control what secrets an authenticated user or application can access.
+- **Audit Log**: Vault maintains detailed logs of all accesses and changes, which is crucial for security audits and compliance.
+
+### Creating Secrets in Hashicorp Vault
+
+HashiCorp Vault configurations can be managed via the Vault CLI. Before you begin configuring Vault, ensure that you have [Vault installed](https://www.vaultproject.io/downloads). After installation, initialize and start the Vault server.
```
Comment (scottslowe) [inline themes/default/content/what-is/what-is-hashicorp-vault.md:26]: There may also be some CLI configuration needed here, but I don't know if you necessarily want to get that deep into the weeds.

File: themes/default/content/what-is/what-is-hashicorp-vault.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+```bash
+# Create a new secret
+vault kv put secret/hello myvalue=s3cr3t
+```
+
+Replace `myvalue=s3cr3t` with the key-value pair you wish to store as a secret.
+
+{{< notes type="info" >}}
+
+The commands provided in this article include `<key>=<value>` parameters to pass secrets to Vault directly through the CLI. Be aware that executing these commands can leave sensitive data in your shell's unencrypted history. Consult the [Static Secrets: Key/Value Secrets Engine tutorial](https://www.vaultproject.io/docs/secrets/kv/kv-v2) for best practices in production scnarios.
```
Comment (scottslowe) [inline themes/default/content/what-is/what-is-hashicorp-vault.md:39]:
```suggestion
The commands provided in this article include `<key>=<value>` parameters to pass secrets to Vault directly through the CLI. Be aware that executing these commands can leave sensitive data in your shell's unencrypted history. Consult the [Static Secrets: Key/Value Secrets Engine tutorial](https://www.vaultproject.io/docs/secrets/kv/kv-v2) for best practices in production scenarios.
```

**Review by scottslowe — COMMENTED** [review]
A few minor comments, but otherwise a solid draft. Well done.

File: themes/default/content/what-is/what-is-hashicorp-vault.md (hunk 6 above)
Comment (interurban) [reply themes/default/content/what-is/what-is-hashicorp-vault.md:26]: Yea; that's been the thinking thus far, not to go far into al the steps, I can add a bit about CLI configuration after installation

**Review by desteves — APPROVED** [review]

**Pull request merged**